cloudbridge/cloud/providers/gce/services.py

```diff
@@ -661,3 +667,196 @@ def routers(self):
 
     def create_router(self, name=None):
         raise NotImplementedError('To be implemented')
+
+
+class GCEBlockStoreService(BaseBlockStoreService):
+
+    def __init__(self, provider):
+        super(GCEBlockStoreService, self).__init__(provider)
+
+        # Initialize provider services
+        self._volume_svc = GCEVolumeService(self.provider)
+        self._snapshot_svc = GCESnapshotService(self.provider)
+
+    @property
+    def volumes(self):
+        return self._volume_svc
+
+    @property
+    def snapshots(self):
+        return self._snapshot_svc
+
+
+class GCEVolumeService(BaseVolumeService):
+
+    def __init__(self, provider):
+        super(GCEVolumeService, self).__init__(provider)
+
+    def get(self, volume_id):
+        """
+        Returns a volume given its id.
+        """
+        try:
+            response = self.provider.get_gce_resource_data(volume_id)
+            if response:
+                return GCEVolume(self.provider, response)
+        except googleapiclient.errors.HttpError as http_error:
+            # If the volume is not found, the API will raise
+            # googleapiclient.errors.HttpError.
+            cb.log.warning(
+                "googleapiclient.errors.HttpError: {0}".format(http_error))
+        return None
+
+    def find(self, name, limit=None, marker=None):
+        """
+        Searches for a volume by a given list of attributes.
+        """
+        filtr = 'name eq ' + name
+        max_result = limit if limit is not None and limit < 500 else 500
+        response = (self.provider
+                        .gce_compute.disks()
+                        .list(project=self.provider.project_name,
+                              zone=self.provider.default_zone,
+                              filter=filtr,
+                              maxResults=max_result,
+                              pageToken=marker).execute())
+        if 'items' not in response:
+            return []
+        gce_vols = [GCEVolume(self.provider, vol)
+                    for vol in response['items']]
+        return ServerPagedResultList(len(gce_vols) > max_result,
+                                     response.get('nextPageToken'),
+                                     False, data=gce_vols)
+
+    def list(self, limit=None, marker=None):
+        """
+        List all volumes.
+        """
+        # For GCE API, Acceptable values are 0 to 500, inclusive.
```

Comment:
It is not completely obvious without going to the API documentation that this comment (and maxResults) are about response paging and not about completely limiting the total number of records returned. In fact it is unclear what the "limit" parameter means- per-page results or total results. Please add comments to clarify.